nifi-nar-bundles/nifi-morphlines-bundle/nifi-morphlines-processors/src/main/java/org/apache/nifi/processors/morphlines/ImplementMorphlines.java

```diff
@@ -0,0 +1,219 @@
+import org.apache.nifi.annotation.lifecycle.OnStopped;
+import org.apache.nifi.processor.exception.*;
+
+@Tags({"kitesdk", "morphlines", "ETL", "HDFS", "avro", "Solr", "HBase"})
+@CapabilityDescription("Implements Morphlines (http://kitesdk.org/docs/1.1.0/morphlines/) framework, which performs in-memory container of transformation commands in oder to perform tasks such as loading, parsing, transforming, or otherwise processing a single record.")
+public class MorphlinesProcessor extends AbstractProcessor {
+
+    private Command morphline;
+    private volatile Record record = new Record();
+    private volatile Collector collector = new Collector();
+
+    public static final PropertyDescriptor MORPHLINES_ID = new PropertyDescriptor
+            .Builder().name("Morphlines ID")
+            .description("Identifier of the morphlines context")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    public static final PropertyDescriptor MORPHLINES_FILE = new PropertyDescriptor
+            .Builder().name("Morphlines File")
+            .description("File for the morphlines context")
+            .required(true)
+            .addValidator(StandardValidators.FILE_EXISTS_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    public static final PropertyDescriptor MORPHLINES_OUTPUT_FIELD = new PropertyDescriptor
+            .Builder().name("Morphlines output field")
+            .description("Field name of output in Morphlines. Default is '_attachment_body'.")
+            .required(false)
+	    .expressionLanguageSupported(true)
+            .defaultValue("_attachment_body")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("Relationship for success.")
+            .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("Relationship for failure of morphlines.")
+            .build();
+
+    private static final List<PropertyDescriptor> PROPERTIES = ImmutableList.<PropertyDescriptor>builder()
+            .add(MORPHLINES_FILE)
+            .add(MORPHLINES_ID)
+            .add(MORPHLINES_OUTPUT_FIELD)
+            .build();
+
+    private static final Set<Relationship> RELATIONSHIPS = ImmutableSet.<Relationship>builder()
+            .add(REL_SUCCESS)
+            .add(REL_FAILURE)
+            .build();
+
+
+    private File morphLinesFile;
+    private String morphLinesId;
+    private String morphlinesOutputField;
+    private PropertyValue morphLinesFileProperty;
+    private PropertyValue morphLinesIdProperty;
+    private PropertyValue morphLinesOutputField;
+    private MorphlineContext morphlineContext;
```

Comment:
You didn't address my comment[1] on these variables. Please do so.

[1] https://github.com/apache/nifi/pull/1529#discussion_r102964711